File: leetcode/data_struct/array/0001/1.ts
// auther: phoenix
// date: 2023-06-08 15:08:43



function twoSum1 (nums:number[], target:number): number[] {
  const len:number = nums.length
  for (let i = 0; i < len; i++) {
    for (let j = i+1; j < (len); j++) {
      if(nums[i] + nums[j] === target) {
        return [i, j]
      }
    }
  }
  return []
}

test()
function test(){
  console.log(twoSum1([2, 5, 9, 7], 9))
  console.log(twoSum1([3, 2, 4], 6))
  console.log(twoSum1([3, 3], 6))
}

File: leetcode/data_struct/array/0035/1.go
// auther: phoenix
// date: 2023-06-12 09:40:11
package main

import "fmt"

/*


 */

func main() {
	searchInsert1Test()
}

func searchInsert1Test() {
	fmt.Println(searchInsert1([]int{1, 3, 5, 6}, 5))
	fmt.Println(searchInsert1([]int{1, 3, 5, 6}, 2))
	fmt.Println(searchInsert1([]int{1, 3, 5, 6}, 7))
	fmt.Println(searchInsert1([]int{1, 3, 5, 6}, 0))
}

func IsContain(items []int, item int) bool {
	for _, eachItem := range items {
		if eachItem == item {
			return true
		}
	}
	return false
}

func searchInsert1(nums []int, target int) int {
	index := 0
	length := len(nums)
	contains := IsContain(nums, target)

	for i := 0; i < length; i++ {
		if contains && nums[i] == target {
			return i
		}
		if !contains && nums[i] <= target {
			index = i + 1
		}
	}

	return index
}
